windows/src/global/delphi/general/CompilePackage.pas

```diff
@@ -282,20 +283,39 @@ function TCompilePackage.BuildKMP: Boolean;
 
     kmpinf.RemoveFilePaths;
 
-    if kmpinf.LexicalModels.Count = 0
-      then kmpinf.Options.FileVersion := SKeymanVersion70
-      else kmpinf.Options.FileVersion := SKeymanVersion120;
+    //
+    // Find the minimum package version based on keyboard version
+    // See also MergeKeyboardInfo.pas
+    //
 
-    if kmpinf.Options.FileVersion = SKeymanVersion70 then
+    if kmpinf.LexicalModels.Count = 0 then
     begin
+      kmpinf.Options.FileVersion := SKeymanVersion70;
+
+      for i := 0 to kmpinf.Keyboards.Count - 1 do
+      begin
+        if CompareVersions(kmpinf.Options.FileVersion, kmpinf.Keyboards[i].MinKeymanVersion) > 0 then
+          kmpinf.Options.FileVersion := kmpinf.Keyboards[i].MinKeymanVersion;
```

Comment:
I see it in the code here. Should we also have the user test verify that if a package contains multiple keyboards, the package fileVersion will be the minimum `MinKeymanVersion` among the keyboards? (e.g. among 7.0, 12.0, 15.0)

Reply:
It should be the maximum, not the minimum; I think the code is clear enough and I don't see an urgent need to user test it -- tester has enough to do as it is I think.